<launch>
  <!-- Arguments -->
  <arg name="model" default="waffle_pi" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="map_file" default="short"/>
  <arg name="open_rviz" default="true"/>
  <arg name="move_forward_only" default="false"/>

  <!-- Turtlebot3 -->
  <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
    <arg name="model" value="$(arg model)" />
  </include>

  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(find ca2_ttk4192)/maps/$(arg map_file)_maze.yaml"/>

  <!-- AMCL -->
  <include file="$(find turtlebot3_navigation)/launch/amcl.launch"/>

  <!-- rviz -->
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find ca2_ttk4192)/rviz/collision_detection.rviz"/>
  </group>

  <node pkg="ca2_ttk4192" type="collision_detection.py" name="collision_detector" output="screen"/>

  <node pkg="ca2_ttk4192" type="position_control.py" name="position_controller" output="screen"/>

</launch>
